time-macros/src/lib.rs

```diff
@@ -89,3 +91,109 @@ pub fn format_description(input: TokenStream) -> TokenStream {
+    let serialize_fns = quote! {
+        pub fn serialize<S: ::serde::Serializer>(
+            datetime: &#(formattable.clone()),
+            serializer: S,
+        ) -> Result<S::Ok, S::Error> {
+            use ::serde::Serialize;
+            datetime
+                .format(&DESCRIPTION)
+                .map_err(::time::error::Format::into_invalid_serde_value::<S>)?
+                .serialize(serializer)
+        }
+
+        pub fn deserialize<'a, D: ::serde::Deserializer<'a>>(
+            deserializer: D
+        ) -> Result<#(formattable.clone()), D::Error> {
+            use ::serde::Deserialize;
+            #(formattable.clone())::parse(<&str>::deserialize(deserializer)?, &DESCRIPTION)
+                .map_err(::time::error::Parse::to_invalid_serde_value::<D>)
+        }
+    };
+    let option_serialize_fns = quote! {
+        pub fn serialize<S: ::serde::Serializer>(
+            option: &Option<#(formattable.clone())>,
+            serializer: S,
+        ) -> Result<S::Ok, S::Error> {
+            use ::serde::Serialize;
+            option.map(|datetime| datetime.format(&DESCRIPTION))
+                    .transpose()
+                    .map_err(::time::error::Format::into_invalid_serde_value::<S>)?
+                    .serialize(serializer)
+        }
+
+        pub fn deserialize<'a, D: ::serde::Deserializer<'a>>(
+            deserializer: D
+        ) -> Result<Option<#(formattable.clone())>, D::Error> {
+            use ::serde::Deserialize;
+            Option::<&str>::deserialize(deserializer)?
+                .map(|string| #(formattable.clone())::parse(string, &DESCRIPTION))
+                .transpose()
+                .map_err(::time::error::Parse::to_invalid_serde_value::<D>)
+        }
+    };
+
+    quote! {
+        mod #(mod_name) {
+            use ::time::#(formattable.clone());
+
+            const DESCRIPTION: &[::time::format_description::FormatItem<'_>] = &[#(items)];
+
+            #(serialize_fns)
+
+            pub mod option {
+                use super::{DESCRIPTION, #(formattable)};
+
+                #(option_serialize_fns)
+            }
+        }
+    }
+}
+
+#[proc_macro]
+pub fn declare_format_string(input: TokenStream) -> TokenStream {
+    let mut tokens = input.into_iter().peekable();
+    // First, an identifier (the desired module name)
+    let mod_name = match tokens.next() {
+        Some(TokenTree::Ident(ident)) => ident,
+        Some(tree) => {
+            return Error::UnexpectedToken { tree }.to_compile_error_standalone();
+        }
+        None => return Error::UnexpectedEndOfInput.to_compile_error_standalone(),
+    };
+    // Followed by a comma
+    if let Err(err) = helpers::consume_punct(',', &mut tokens) {
+        return err.to_compile_error_standalone();
+    }
+    // Then, the type to create serde serializers for (e.g., `OffsetDateTime`).
+    let formattable = match tokens.next() {
+        Some(TokenTree::Ident(ident)) => TokenTree::Ident(ident),
```

Comment:
This should be able to be replaced with `Some(tree @ TokenTree::Ident(_)) => tree`, which avoids creating a new value.

Reply:
Nice, that's a new trick for me :)